tree-sitter-darklang/grammar.js

```diff
@@ -874,8 +903,25 @@ module.exports = grammar({
 
     qualified_type_name: $ =>
       seq(
-        repeat(seq($.module_identifier, alias(".", $.symbol))),
-        $.type_identifier,
+        seq(
+          repeat(seq($.module_identifier, alias(".", $.symbol))),
+          field("type_identifier", $.type_identifier),
+        ),
+        optional(
+          seq(
+            field("symbol_open_angle", alias("<", $.symbol)),
+            field("type_args", $.type_args),
+            field("symbol_close_angle", alias(">", $.symbol)),
+          ),
+        ),
+      ),
+
+    type_args: $ =>
+      seq(
+        $.type_reference,
+        repeat(
+          seq(field("symbol_comma", alias(",", $.symbol)), $.type_reference),
+        ),
       ),
```

Comment:
Separate from my other long comment: I do think that `type_args` should include the `<` symbols.